pkg/runtime/wfengine/workflow.go

```diff
@@ -578,7 +604,24 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 	state.ApplyRuntimeStateChanges(runtimeState)
 	state.ClearInbox()
 
-	return wf.saveInternalState(ctx, actorID, state)
+	err = wf.saveInternalState(ctx, actorID, state)
+	if err != nil {
+		return err
+	}
+	if executionStatus != "" {
```

Comment:
isn't the only possible option here is that executionStatus is diag.StatusFailed here? Can we check on that? Or do we have other possible values which are then being reset in line below when it is set back to empty string?

Reply:
No, executionStatus can be empty as well, which happens when workflow has already completed beforehand: https://github.com/dapr/dapr/pull/7152/files#diff-9997677e8aa2146ca3f09e604ca0f932a17a85fddc758fcc785c7ebb9afbc08aR457

for e.g. If a terminate signal is sent to a workflow and it has already completed, we don't want to re-record the metrics and so executionStatus would be empty here and so it won't be overwritten in the logic below and therefore in the defer statement, the metrics won't be recorded.